contracts/MechMarketplace.sol

```diff
@@ -194,21 +169,26 @@ contract MechMarketplace is IErrorsMarketplace {
             revert ZeroValue();
         }
 
+        // TODO What to do if there is rateDIff leftovers? Keep as a fee or record into sender's map?
+        uint256 rateDiff;
+        if (actualDeliveryRate > deliveryRate) {
+            rateDiff = actualDeliveryRate - deliveryRate;
+        }
+
         // TODO what if fee is zero just because the delivery rate is in the order of 1..10_000?
         // Calculate mech payment and marketplace fee
-        marketplaceFee = (deliveryRate * fee) / 10_000;
-        mechPayment = deliveryRate - marketplaceFee;
+        marketplaceFee = (actualDeliveryRate * fee) / 10_000;
+        mechPayment = actualDeliveryRate - marketplaceFee;
```

Comment:
DO NOT apply this here. rounding issues will render it unusable. Can only apply take rate at withdrawal for large (lower bound enforced!) amounts